sdk/FunctionMetadataLoaderExtension/Startup.cs

```diff
@@ -50,42 +52,63 @@ public void Configure(IWebJobsBuilder builder)
             // This will not be called.
         }
 
-        internal static string GetPathToExe(string appRootPath)
+        internal static string GetUpdatedExeIfSelfContained(string appRootPath)
         {
-            string fullPathToWorkerConfig = Path.Combine(appRootPath, _workerConfigFile);
+            string fullPathToWorkerConfig = Path.Combine(appRootPath, WorkerConfigFile);
+            string? exeValue = GetExeValueFromWorkerConfig(fullPathToWorkerConfig);
 
-            if (!File.Exists(fullPathToWorkerConfig))
+            if (HasSelfContainedIdentifier(exeValue))
```

Comment:
Another nit -- but could you do `string.Replace()` here? If there's no `{WorkerRoot}`, it'll be a no-op.